### PR TITLE
Refactor file system

### PR DESCRIPTION
**Pull request by siyuan0322**
Take advantage of [fsspec](https://github.com/intake/filesystem_spec) to resolve common schemes.
Implement a customized oss driver.

### CODE REVIEW

File: modules/graph/loader/arrow_fragment_loader.h [hunk 1]
```diff
@@ -189,10 +189,9 @@ GatherETables(Client& client,
     }
     std::unordered_map<std::string, std::string> meta_map;
     metadata->ToUnorderedMap(&meta_map);
-    grouped_batches[meta_map.at("label")]
-                   [std::make_pair(meta_map.at("src_label"),
-                                   meta_map.at("dst_label"))]
-                       .emplace_back(batch);
+    grouped_batches[meta_map["label"]][std::make_pair(meta_map.at("src_label"),
+                                                      meta_map.at("dst_label"))]
+        .emplace_back(batch);
```
Comment (sighingnow) [inline modules/graph/loader/arrow_fragment_loader.h:194]: Are `"src_label"` and `"dst_label"` guaranteed to exist?

File: modules/io/CMakeLists.txt [hunk 2]
```diff
@@ -118,6 +118,10 @@ foreach(fname ${PYTHON_IO_ADAPTORS})
     add_dependencies(vineyard_io_adaptors ${IO_BINARY_TOOL})
 endforeach()
 
+install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/adaptors/ossfs.py
+        DESTINATION bin
+)
```
Comment (sighingnow) [inline modules/io/CMakeLists.txt:123]: Why?

File: modules/io/python/requirements.txt [hunk 3]
```diff
@@ -2,3 +2,5 @@ hdfs3
 pyarrow
 pyorc
 vineyard
+fsspec
+s3fs
```
Comment (sighingnow) [inline modules/io/python/requirements.txt:6]: Sort dependencies in alphabetical order.

**Review by sighingnow — APPROVED** [review]
I suggest to submit a code-format PR first, using any formatter you would like. Such a code-format doesn't involves any semantic change.

Then rebase/reset/apply this PR on that.

File: modules/graph/loader/arrow_fragment_loader.h (hunk 1 above)
Comment (siyuan0322) [reply modules/graph/loader/arrow_fragment_loader.h:194]: These three key may not exists, however, a empty string will be created automatically and that will suffice.

Comment (sighingnow) [reply modules/graph/loader/arrow_fragment_loader.h:194]: `at` will raise.

**Pull request merged**